# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/WindowGlobalParent.h
# Commit: 2c4f84b0e661
# Full Hash: 2c4f84b0e66164503066908c973fef983623c864
# Author: Tim Huang <tihuang@mozilla.com>
# Date: 2020-01-14 03:45:02
# Regressor Bug: 1599043
# File Overlap Count: 1
# Description:
#   Bug 1599043 - Part 1: Add a ContentBlockingLog into the WindowGlobalParent r=dimi,Ehsan
#   
#   This adds a ContentBlockingLog into the WindowGlobalParent. This
#   ContentBlockingLog is bascially a copy of the log in the content
#   process. This log in parent is needed for the OnContentBlockingEvent in
# ==============================================================================

diff -r b54a7ff4c097 -r 2c4f84b0e661 dom/ipc/WindowGlobalParent.h
--- a/dom/ipc/WindowGlobalParent.h	Mon Jan 13 14:07:09 2020 +0000
+++ b/dom/ipc/WindowGlobalParent.h	Mon Jan 13 14:06:50 2020 +0000
@@ -7,6 +7,7 @@
 #ifndef mozilla_dom_WindowGlobalParent_h
 #define mozilla_dom_WindowGlobalParent_h
 
+#include "mozilla/AntiTrackingCommon.h"
 #include "mozilla/RefPtr.h"
 #include "mozilla/dom/DOMRect.h"
 #include "mozilla/dom/PWindowGlobalParent.h"
@@ -16,6 +17,7 @@
 #include "nsISupports.h"
 #include "mozilla/dom/WindowGlobalActor.h"
 #include "mozilla/dom/CanonicalBrowsingContext.h"
+#include "mozilla/dom/ContentBlockingLog.h"
 
 class nsIPrincipal;
 class nsIURI;
@@ -130,6 +132,14 @@
   JSObject* WrapObject(JSContext* aCx,
                        JS::Handle<JSObject*> aGivenProto) override;
 
+  void NotifyContentBlockingEvent(
+      uint32_t aEvent, nsIRequest* aRequest, bool aBlocked, nsIURI* aURIHint,
+      const nsTArray<nsCString>& aTrackingFullHashes,
+      const Maybe<AntiTrackingCommon::StorageAccessGrantedReason>& aReason =
+          Nothing());
+
+  ContentBlockingLog* GetContentBlockingLog() { return &mContentBlockingLog; }
+
  protected:
   const nsAString& GetRemoteType() override;
   JSWindowActor::Type GetSide() override { return JSWindowActor::Type::Parent; }
@@ -178,6 +188,11 @@
 
   // True if this window has a "beforeunload" event listener.
   bool mHasBeforeUnload;
+
+  // The log of all content blocking actions taken on the document related to
+  // this WindowGlobalParent. This is only stored on top-level documents and
+  // includes the activity log for all of the nested subdocuments as well.
+  ContentBlockingLog mContentBlockingLog;
 };
 
 }  // namespace dom
